testing/e2e/app/routes/Menu/route.tsx

```diff
@@ -0,0 +1,83 @@
+import { useSearchParams } from '@remix-run/react';
+import { Menu } from 'node_modules/@itwin/itwinui-react/esm/core/Menu/Menu.js';
```

Comment:
e2e tests should only test the public API surface. The `Menu` component is a private implementation detail of `DropdownMenu` and other components, so prefer testing those instead.

Reply:
> I thought about moving these `Menu` tests to `DropdownMenu` e2e tests but felt that they were not the right place for them since some tests are `Menu` specific and `Menu` is used in other components too apart from just the `DropdownMenu`.

I don't agree with this logic. `DropdownMenu` is a very thin wrapper over `Menu` so it's the best place to test it. If it helps, you can think of it as adding new tests to verify that `DropdownMenu` is behaving correctly. `Menu` is an implementation that literally does not matter.

> The only `Menu` e2e test now is one that checks that the focus moves correctly when opening/closing the `Menu` with a keyboard and mouse.

This needs to be removed. We should not have *any* e2e tests that directly use `Menu`.

Reply:
I'm ok with just testing `DropdownMenu`, but you can optionally also test the other components if you feel it's useful.

Reply:
The e2e tests can be improved and cleaned up a bit.
- Instead of rendering multiple menus, you can create local components (`<DropdownMenuWithSubmenus>`, `<DropdownMenuWithExtraContent>`) that can be conditionally rendered based on search params (see other e2e tests).
- Some of the test `expect`s are missing `await`s.
  - This probably causes flakiness which need to be resolved using hacks like the `keyboardPressOptions` object.
- Some of the test names don't match what is being tested. e.g. the "should respect click and keyboard enter/space triggers" doesn't test for clicks anymore.
  - Further, manually calling `.focus()` doesn't exactly match how the user would interact with it. The user would <kbd>Tab</kbd> to the button. Maybe this is ok though.